WordPress/Classes/ViewRelated/Blog/SiteSettingsViewController.m

```diff
@@ -472,6 +516,62 @@ - (void)tableView:(UITableView *)tableView didSelectInAccountSectionRow:(NSInteg
     }
 }
 
+- (void)showPostFormatSelector
+{
+    NSArray *titles = self.blog.sortedPostFormatNames;
+    NSArray *formats = self.blog.sortedPostFormats;
+    if (titles.count == 0 || self.blog.defaultPostFormatText == nil) {
+        return;
+    }
+    NSString *currentDefaultPostFormat = self.blog.defaultPostFormat;
+    if (!currentDefaultPostFormat) {
+        currentDefaultPostFormat = formats[0];
+    }
+    NSDictionary *postFormatsDict = @{
+                                      SettingsSelectionDefaultValueKey   : [formats firstObject],
+                                      SettingsSelectionTitleKey          : NSLocalizedString(@"Default Post Format", @"Title for screen to select a default post format for a blog"),
+                                      SettingsSelectionTitlesKey         : titles,
+                                      SettingsSelectionValuesKey         : formats,
+                                      SettingsSelectionCurrentValueKey   : currentDefaultPostFormat
+                                      };
+    
+    PostSettingsSelectionViewController *vc = [[PostSettingsSelectionViewController alloc] initWithDictionary:postFormatsDict];
+    __weak __typeof__(self) weakSelf = self;
+    vc.onItemSelected = ^(NSString *status) {
+        // Check if the object passed is indeed an NSString, otherwise we don't want to try to set it as the post format
+        if ([status isKindOfClass:[NSString class]]) {
+            weakSelf.blog.defaultPostFormat = status;
+            [weakSelf saveSettings];
+        }
+    };
+    
+    [self.navigationController pushViewController:vc animated:YES];
+}
+
+- (void)tableView:(UITableView *)tableView didSelectInWritingSectionRow:(NSInteger)row
+{
+    switch (row) {
+        case SiteSettingsWritingDefaultCategory:{
+            PostCategoryService *postCategoryService = [[PostCategoryService alloc] initWithManagedObjectContext:[[ContextManager sharedInstance] mainContext]];
+            NSNumber *defaultCategoryID = self.blog.defaultCategory;
+            if (!defaultCategoryID) {
+                defaultCategoryID = @(1);
+            }
+            PostCategory *postCategory = [postCategoryService findWithBlogObjectID:self.blog.objectID andCategoryID:defaultCategoryID];
+
+            PostCategoriesViewController *postCategoriesViewController = [[PostCategoriesViewController alloc] initWithBlog:self.blog
+                                                                                                           currentSelection:@[postCategory]
+                                                                                                              selectionMode:CategoriesSelectionModeBlogDefault];
```

Comment:
I just got a crash here. I logged in and went to blog settings, when I tapped on Default Category it crashed. I guess the list of categories wasn't synced yet, so `postCategory == nil`, and `@[nil]` crashes